src/data/data1d/filter1d/fft.ts

```diff
@@ -60,3 +69,15 @@ function generateXAxis(datum1D) {
   }
   return xAxis;
 }
+
+function isPowerOfTwo(n) {
+  return n !== 0 && (n & (n - 1)) === 0;
+}
+
+function hasSameLength(data: Data1D) {
+  const xLength = data.x.length;
```

Comment:
Why using a loop (and even a function) just to do:

(data.x.length === data.re.length) && (data.x.length === data.im.length)

or did I miss something.

Reply:
yes, I remove the function and did it directly as the conditional